tests/covalent_tests/executor/executor_plugins/remote_executor_test.py

```diff
@@ -0,0 +1,161 @@
+# notice, and modified files must contain a notice indicating that they have
+# been altered from the originals.
+#
+# Covalent is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the License for more details.
+#
+# Relief from the License may be granted by purchasing a commercial license.
+
+"""Tests for Covalent remote executor."""
+
+import tempfile
+from typing import Dict
+
+import pytest
+
+from covalent.executor.executor_plugins.remote_executor import RemoteExecutor
+
+
+class MockRemoteExecutor(RemoteExecutor):
+    def __init__(self):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            tmpfile = tempfile.NamedTemporaryFile(
+                prefix="mock_credentials", suffix=".rsa", dir=tmpdir
+            )
+
+        super().__init__(poll_freq=10, remote_cache=tmpdir, credentials_file=tmpfile)
+
+    async def run(self, function, args, kwargs, task_metadata):
+        return await function(*args, **kwargs)
+
+    async def _validate_credentials(self):
+        pass
+
+    async def _upload_task(self):
+        pass
+
+    async def submit_task(self, task_metadata: Dict):
+        pass
+
+    async def _poll_task(self):
+        pass
+
+    async def get_status(self):
+        pass
+
+    async def query_result(self):
+        pass
+
+    async def cancel(self):
+        return "workflow cancel request succeeded"
+
+
+def test_remote_executor_init():
+    """Test remote executor constructor"""
+
+    mock_executor = MockRemoteExecutor()
+    assert mock_executor.poll_freq == 10
+    assert "mock_credentials" in mock_executor.credentials_file.name.split("/")[-1].split(".")[0]
+    assert "rsa" in mock_executor.credentials_file.name.split("/")[-1].split(".")[-1]
+
+
+@pytest.mark.asyncio
+async def test_run_async_subprocess():
+    """Test remote executor async subprocess call"""
+
+    test_dir, test_file, non_existent_file = "file_dir", "file.txt", "non_existent_file.txt"
+    create_file = (
+        f"rm -rf {test_dir} && mkdir {test_dir} && cd {test_dir} && touch {test_file} && echo 'hello remote "
+        f"executor' >> {test_file} "
+    )
+    read_non_existent_file = f"cat {non_existent_file}"
+
+    executor = MockRemoteExecutor()
+    create_file_proc, create_file_stdout, create_file_stderr = await executor.run_async_subprocess(
+        create_file
+    )
+
+    # Test that file creation works as expected
+    assert create_file_proc.returncode == 0
+    assert create_file_stdout.decode() == ""
+    assert create_file_stderr.decode() == ""
+
+    # Test that file was created and written to correctly
+    try:
+        with open(f"{test_dir}/{test_file}", "r") as test_file:
+            lines = test_file.readlines()
+            assert lines[0].strip() == "hello remote executor"
+
+    except FileNotFoundError as fe:
+        return str(fe)
```

Comment:
if `FileNotFoundError` is raised and this is indicative that the test has failed, let's replace the `return` statement with a `pytest.fail()` call with a descriptive message.

Reply:
I have replaced the statement => https://github.com/AgnostiqHQ/covalent/pull/1177/commits/aaf90bfe36a2b598b18728e09f593cd17213059a